File: compiler/cpp/src/exception/MaxLookaheadException.cpp
/*
Copyright © 2021-2022 Compose Generator Contributors
All rights reserved.
*/

/*
Copyright © 2021 Compose Generator Contributors
All rights reserved.
*/

#include "MaxLookaheadException.h"

const char *MaxLookaheadException::what() const noexcept {
    return errorMessage.c_str();
}

File: compiler/java/src/main/java/com/chillibits/ccom/reader/MaxLookAheadException.java
/*
Copyright © 2021-2022 Compose Generator Contributors
All rights reserved.
*/

package com.chillibits.ccom.reader;

/**
 * Exception indicating that the max look ahead is less than 1.
 */
public class MaxLookAheadException extends Exception {

    public MaxLookAheadException(int maxLookAhead) {
        super("Max look ahead was " + maxLookAhead + " but must be at least 1");
    }

}
